app-editors/scite/scite-4.1.2.ebuild

```diff
@@ -0,0 +1,156 @@
+# Copyright 1999-2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit flag-o-matic gnome2-utils llvm toolchain-funcs xdg-utils
+
+MY_PV=${PV//./}
+DESCRIPTION="A very powerful, highly configurable, small editor with syntax coloring"
+HOMEPAGE="http://www.scintilla.org/SciTE.html"
+SRC_URI="https://www.scintilla.org/${PN}${MY_PV}.tgz -> ${P}.tgz"
+
+LICENSE="HPND lua? ( MIT )"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~x86 ~x86-fbsd ~amd64-linux ~arm-linux ~x86-linux"
+IUSE="lua clang"
+
+RDEPEND="dev-libs/glib:=
+	x11-libs/cairo
+	x11-libs/gdk-pixbuf
+	x11-libs/gtk+:3=
+	x11-libs/pango
+	lua? ( >=dev-lang/lua-5:= )
+	clang? ( >=sys-devel/clang-5:= )
+"
+
+DEPEND="${RDEPEND}
+	virtual/pkgconfig"
+
+S="${WORKDIR}/${PN}/gtk"
+
+pkg_pretend() {
+	# use clang could be package specific
+	if use clang && !(tc-is-clang) ; then
+		local clangMajorVersion="$( clang++ -dM -E -x c /dev/null |
+			 grep __clang_major | grep -o -E '[[:digit:]]' )"
+
+		# need C++17 features
+		# FIXME: Better Check of C++17 Features
+		[[ "${MERGE_TYPE}" != "binary" &&
+		${clangMajorVersion} -lt 5 ]] &&
+		die "Sorry, SCiTE uses C++17 Features and needs >sys-devel/clang-5
+		($(clangMajorVersion))."
+
+	# global choice of clang
+	elif tc-is-clang ; then
+		# need c++17 features
+		[[ "${MERGE_TYPE}" != "binary" &&
+		$(clang-major-version) -lt 5 ]] &&
+		die "Sorry, SCiTE uses C++17 Features and needs >sys-devel/clang-5
+		($(clang-major-version))."
+
+	# standard: gcc
+	elif tc-is-gcc; then
+		# older gcc is not supported
+		[[ "${MERGE_TYPE}" != "binary" &&
+		$(gcc-major-version) -lt 7 ]] &&
+		die "Sorry, Scite uses C++17 Features, need >sys-devel/gcc-7."
+	else
+		die "Either gcc or clang should be available for building scite"
+	fi;
+}
+
+pkg_setup() {
+	# presumably needed and maybe useful
+	[ use clang || tc-is-clang ] && llvm_pkg_setup;
```

Comment:
This is good, the syntax in here is simpler though: `use clang && llvm_pkg_setup`

In future always use `[[ ]]` double brackets with bash.